src/app/(main)/(auth)/sign-up/[[...sign-up]]/sign-up-form.tsx

```diff
@@ -0,0 +1,125 @@
+"use client";
+
+import { AuthForm, authFormSchema } from "@/app/(main)/(auth)/auth-schemas";
+import { Alert, AlertTitle, AlertDescription } from "@/components/ui/alert";
+import { Button } from "@/components/ui/button";
+import { Form, FormControl, FormField, FormItem, FormLabel, FormMessage } from "@/components/ui/form";
+import { Input } from "@/components/ui/input";
+import { useSignUp } from "@clerk/nextjs";
+import { zodResolver } from "@hookform/resolvers/zod";
+import { AlertCircle } from "lucide-react";
+import { useState } from "react";
+import { useForm } from "react-hook-form";
+import { finishSignUpAction } from "../../auth-actions";
+
+export function SignUpForm() {
+  const { isLoaded, signUp, setActive } = useSignUp();
+  const [errorMessage, setErrorMessage] = useState<null | string>(null);
+  const [verifying, setVerifying] = useState(false);
+  const form = useForm<AuthForm>({
+    resolver: zodResolver(authFormSchema),
+    defaultValues: {
+      email: "",
+      password: "",
+      verificationCode: "000000",
```

Comment:
I think we don't have to include all props in default values so something like

```
defaultValues: {
      email: "",
      password: "",
}
```

should work

Reply:
That was specifically to allow the first submit before the email code is sent, the zeros satisfying the zod check, then it gets reset to "" before actually accepting the verification code